docs/source/using-discovery/slurm.md

```diff
@@ -0,0 +1,752 @@
+Remember, troubleshooting requires patience and a systematic approach. Begin by identifying the problem, then hypothesize potential causes, test those hypotheses, and apply solutions. Make small changes one at a time and retest after each change. With experience, you will be able to troubleshoot effectively and make the most of your HPC resources.
+
+## Best Practices
+In this section, we will discuss some best practices that can help you make efficient use of resources, write optimized job scripts, and ensure maximum throughput and minimal queue time in a Slurm-based HPC environment.
+
+### Efficient Usage of Resources
+1. **Request Only What You Need:** When submitting a job, request only the resources that you need. Overestimating your requirements can result in longer queue times as Slurm waits for the requested resources to become available.
+1. **Use Job Arrays for Similar Jobs:** If you need to run multiple similar jobs, consider using job arrays. This approach makes job management more straightforward and reduces overhead.
+1. **Use Appropriate Partitions:** Select the appropriate partition for your job based on your requirements. Each partition may have different limits and priorities, so choose the one that suits your needs best.
+
+## Writing Optimized Job Scripts
+1. **Use Environment Variables:** Slurm provides several environment variables that can be used to customize job behavior dynamically. Use these variables to make your scripts more flexible and efficient.
+1. **Specify All Necessary Options:** Ensure to specify all necessary SBATCH options in your job script. Missing options can lead to unpredictable job behavior or performance.
+1. **Check Exit Codes:** Always check the exit codes of commands in your job script. Non-zero exit codes usually indicate an error, and failing to check these can lead to undetected job failures.
+
+:::{code} bash
+command
+if [ $? -ne 0 ]; then
+  echo "Command failed"
+  exit 1
+fi
+:::
+
+### Guidelines to Ensure Maximum Throughput and Minimal Queue Time
+1. **Monitor Job Performance:** Regularly monitor your jobs' performance using `sstat` and adjust resource requests as needed. This can help improve job scheduling efficiency.
+1. **Use the `time` Option Judiciously:** While it's crucial to give your job enough time to complete, overestimating can lead to longer queue times. Start with a reasonable estimate and adjust based on actual run times.
+1. **Prioritize Critical Jobs:** If you have a critical job that needs to be run immediately, you can temporarily increase its priority using `scontrol`. However, use this feature sparingly to maintain fairness.
+
+Remember, these best practices aim to ensure efficient and fair usage of shared HPC resources. Always be mindful of other users and try to use the resources in a way that maximizes productivity for everyone.
+
+
+### Proper resource request syntax
```

Comment:
I think this is good information but it just feels a little out of place here since it is right at the end.

Reply:
This should have been in the best practice subsection. I fixed it so it formats properly.